.github/workflows/ci.yml

```diff
@@ -167,18 +167,18 @@ jobs:
       - uses: actions-rs/toolchain@v1
         with:
           toolchain: ${{ env.nightly }}
+          components: miri
           override: true
       - uses: Swatinem/rust-cache@v1
-      - name: Install Miri
-        run: |
-          set -e
-          rustup component add miri
-          cargo miri setup
-          rm -rf tokio/tests
-
       - name: miri
-        run: cargo miri test --features rt,rt-multi-thread,sync task
+        # Many of tests in tokio/tests and doctests use #[tokio::test] or
+        # #[tokio::main] that calls epoll_create1 that Miri does not support.
+        run: cargo miri test --features fs,io-util,io-std,net,parking_lot,rt,rt-multi-thread,sync --lib --no-fail-fast
```

Comment:
This is opt-ing out of some things that use `tokio::sync::Notified`. Getting intrusive linked lists on the stack to pass with miri will probably be difficult.

Reply:
Or maybe not .. it's possible the problem was elsewhere entirely.

Reply:
Not sure how to fix the timer, but `Notified` works now.

Reply:
Haha, I figured it out: The issue for timers was `futures::executor::block_on`.

Reply:
> The issue for timers was `futures::executor::block_on`.

Ah, `futures` has a few unreleased fixes of stacked borrows violation with `-Zmiri-tag-raw-pointers`: https://github.com/rust-lang/futures-rs/pull/2550